google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -422,12 +425,38 @@ public Bucket update(BucketInfo bucketInfo, BucketTargetOption... options) {
 
   @Override
   public Blob update(BlobInfo blobInfo, BlobTargetOption... options) {
-    return todo();
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+    GrpcCallContext grpcCallContext = GrpcRequestMetadataSupport.create(optionsMap);
+    Object object = codecs.blobInfo().encode(blobInfo);
+    UpdateObjectRequest.Builder updateRequestBuilder =
+        UpdateObjectRequest.newBuilder().setObject(object);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_GENERATION_MATCH),
+        updateRequestBuilder::setIfGenerationMatch);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_GENERATION_NOT_MATCH),
+        updateRequestBuilder::setIfGenerationNotMatch);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_METAGENERATION_MATCH),
+        updateRequestBuilder::setIfMetagenerationMatch);
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.IF_METAGENERATION_NOT_MATCH),
+        updateRequestBuilder::setIfMetagenerationNotMatch);
+    ifNonNull(
+        (String) optionsMap.get(StorageRpc.Option.PREDEFINED_ACL),
+        updateRequestBuilder::setPredefinedAcl);
+    updateRequestBuilder.setUpdateMask(fieldMaskGenerator(object.getAllFields()));
```

Comment:
This line will need to change after the small refactor to the helper method.

```suggestion
    updateRequestBuilder.setUpdateMask(fieldMaskGenerator(object));
```